enterprise/frontend/src/metabase-enterprise/data_editing/tables/edit/use-table-action-form-description.ts

```diff
@@ -8,24 +8,39 @@ type UseTableDescribeTmpModalProps = {
   actionId: BuiltInTableAction;
   scope: TableEditingScope;
   skip?: boolean;
+  refetchDependencies?: unknown[];
 };
 
 export const useActionFormDescription = ({
   actionId,
   scope,
   skip,
+  refetchDependencies = [],
 }: UseTableDescribeTmpModalProps) => {
+  const [shouldFetch, setShouldFetch] = useState(!skip);
   const [fetchModalDescription, { data, isLoading }] =
     useDescribeActionFormMutation();
 
+  // Request re-fetch when `refetchDependencies` or `skip` changes
   useEffect(() => {
-    if (!data && !skip) {
+    if (!skip) {
+      setShouldFetch(true);
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [skip, ...refetchDependencies]);
+
+  // Fetch modal description when `shouldFetch` is true and not currently loading
+  useEffect(() => {
+    if (shouldFetch && !isLoading) {
```

Comment:
lgtm, although I'm more a fan of creating explicit `refetch` callback and providing it from the hook to call near the source of actual dependencies change (similar to how rtk-query/tanstack-query do).